lib/src/repository.rs

```diff
@@ -44,9 +46,13 @@ impl OmicronRepo {
         repo_path: &Utf8Path,
         system_version: Version,
         keys: Vec<Key>,
+        root: Option<SignedRole<Root>>,
```

Comment:
Instead of this, what do you think about always passing in a root and having the caller create it if it's missing? Especially since you made `new_root` public.

Otherwise, would prefer this not be an `Option` since that can be misinterpreted in a few ways. An explicit enum with a `FromKeysAndExpiry` variant would be reasonable here I think.

Reply:
Took the first suggestion.